video/out/vo_gpu_next.c

```diff
@@ -990,16 +990,21 @@ static void draw_frame(struct vo *vo, struct vo_frame *frame)
         p->last_id = id;
     }
 
+    bool pass_colorspace = false;
+    struct pl_color_space hint;
     if (p->next_opts->target_hint && frame->current) {
-        struct pl_color_space hint = frame->current->params.color;
+        hint = frame->current->params.color;
+        if (p->ra_ctx->fns->pass_colorspace && p->ra_ctx->fns->pass_colorspace(p->ra_ctx))
```

Comment:
> I think this should be changed so that pass is only attempted if this function is called, when target-colorspace-hint is set, and don't attempt otherwise. This makes the behavior the same as pl_swapchain_colorspace_hint.

I think it doesn't make sense to make it conditional if it is already based on `target_colorspace` structure. Also it makes little sense to not signal colorspace. `target-colorspace-hint` should be removed, but that's outside of the scope of this PR. It is job for the player to render and send image in proper colorspace and metadata. IMHO sRGB fallback is ancient artifact, many monitors doesn't even have sRGB mode, so it is wrong.

Reply:
The idea behind the naming was that vo_gpu_next passes the colorspace directly to the backend without going through libplacebo. Hence `pass_colorspace`. If you have a better name, I don't mind changing it.

Also it should already only be attempted if `--target-colorspace-hint` is set? Or am I misunderstanding what you mean here?

Reply:
>vo_drm_set_hdr_metadata is always called

Sure but it should effectively be a no-op if you aren't using `--target-colorspace-hint`.

Reply:
>This can change display behavior.

Does it? I would expect drivers to no-op the hdr metadata parameters if you aren't trying to use the HDR mode. It would be pretty broken if they actually tried to do anything with it. But OK it doesn't hurt to be paranoid I guess.

>vo_gpu

This has many problems in this area that I'm not going to bother fixing.

Reply:
Hmm but this proposed change is actually pretty annoying in reality to implement because it messes up the "restore back to sdr" logic.

Reply:
@na-na-hi: Yeah the additional code complexity is not worth it IMO unless you are aware of an actual concrete problem. Setting the metadata blob with SDR works exactly as expected for me. Setting the blob whenever the params change from the VO is the cleanest and best way to handle it. If you want to make vo_gpu aware of various `--target` options, that's fine but I'm not going to do it in this patch.